### PR TITLE
Changed zerostate to be child of replicasetlist state

### PR DESCRIPTION
**Pull request by floreks**
Refactored zerostate and made it child of replicasetlist state. Related to issue #235.

@maciaszczykm @bryk can you review?

### CODE REVIEW

File: src/app/frontend/replicasetlist/replicasetlist_stateconfig.js [hunk 1]
```diff
@@ -24,16 +25,28 @@ import ReplicaSetListController from './replicasetlist_controller';
...
+      .state(zerostate, {
+        views: {
+          '@': {
+            controller: ZeroStateController,
+            controllerAs: 'ctrl',
+            templateUrl: 'replicasetlist/zerostate/zerostate.html',
+          },
+        },
+        url: '/zerostate',
+      });
```
Comment (floreks) [inline src/app/frontend/replicasetlist/replicasetlist_stateconfig.js]: I really don't like formatting of this part :)

Comment (bryk) [inline src/app/frontend/replicasetlist/replicasetlist_stateconfig.js]: Well, at least it is machine formatted and we cannot argue over this :)

It looks a lot better when split into two states:

```js
$stateProvider.state(replicasets, {
    controller: ReplicaSetListController,
    controllerAs: 'ctrl',
    url: replicasetsUrl,
    resolve: {
      'replicaSets': resolveReplicaSets,
    },
    templateUrl: 'replicasetlist/replicasetlist.html',
    onEnter: redirectIfNeeded,
  });
  $stateProvider.state(zerostate, {
    views: {
      '@': {
        controller: ZeroStateController,
        controllerAs: 'ctrl',
        templateUrl: 'replicasetlist/zerostate/zerostate.html',
      },
    },
    url: '/zerostate',
  });
```

File: src/test/frontend/zerostate/zerostate_controller_test.js [hunk 2]
```diff
@@ -12,9 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import ZerostateController from 'zerostate/zerostate_controller';
+import ZerostateController from 'replicasetlist/zerostate/zerostate_controller';
 
-describe('Main controller', () => {
+describe('Zerostate controller', () => {
```
Comment (bryk) [inline src/test/frontend/zerostate/zerostate_controller_test.js]: Should you move this test to replicasetlist?

**Comment by bryk** [comment]
Looks very nice :) Two style comment.

File: src/test/frontend/zerostate/zerostate_controller_test.js (hunk 2 above)
Comment (floreks) [inline src/test/frontend/zerostate/zerostate_controller_test.js]: You mean move folder so structure will be the same as src?

File: src/app/frontend/replicasetlist/replicasetlist_stateconfig.js (hunk 1 above)
Comment (floreks) [inline src/app/frontend/replicasetlist/replicasetlist_stateconfig.js]: Yeah, this way it looks good.

**Comment by maciaszczykm** [comment]
LGTM

**Comment by floreks** [comment]
PTAL

**Comment by bryk** [comment]
LGTM

**Comment by bryk** [comment]
@floreks Hey, something bad happened. I the rename of integration test didn't get merged. Can you send a PR with the rename?

**Comment by floreks** [comment]
I haven't changed integration tests :) Only frontend. Do you want to move frontend zerostate to replicasetlist/zerostate?

**Comment by bryk** [comment]
You're correct :) No need for a change. I just got confused, ignore this.

**Pull request merged**